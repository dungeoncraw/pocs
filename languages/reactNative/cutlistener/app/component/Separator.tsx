import React from 'react';
import { View, StyleSheet, ViewProps } from 'react-native';

interface SeparatorProps extends ViewProps {
    testID?: string;
}

const Separator = ({ testID = 'separator', ...props }: SeparatorProps) => (
    <View style={styles.line} testID={testID} {...props} />
);

const styles = StyleSheet.create({
    line: {
        height: 1,
        backgroundColor: '#014898',
        marginVertical: 10,
        width: '100%',
    },
});

export default Separator;
