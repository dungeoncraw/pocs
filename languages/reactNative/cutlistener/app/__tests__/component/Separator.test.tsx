import React from 'react';
import { render } from '@testing-library/react-native';
import Separator from '@/app/component/Separator';

describe('Separator', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<Separator />);

    // Snapshot test to verify the component renders correctly
    expect(toJSON()).toMatchSnapshot();
  });

  it('has the correct style', () => {
    const { getByTestId } = render(<Separator testID="separator" />);

    const separator = getByTestId('separator');

    // Check that the separator has the correct style
    expect(separator.props.style).toMatchObject({
      height: 1,
      backgroundColor: '#014898',
      marginVertical: 10,
      width: '100%',
    });
  });
});
