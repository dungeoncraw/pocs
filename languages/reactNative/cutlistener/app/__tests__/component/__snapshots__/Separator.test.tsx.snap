// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#014898",
      "height": 1,
      "marginVertical": 10,
      "width": "100%",
    }
  }
  testID="separator"
/>
`;
